gulpfile.js

```diff
@@ -39,80 +47,130 @@ gulp.task('lint', function() {
       `${APP_ROOT}**/*.{js,jsx}`,
       `${TESTS_ROOT}**/*.js`,
       './*.js',
-      `!${APP_ROOT}components/**`,
     ])
     .pipe(eslint())
     .pipe(eslint.format())
     .pipe(eslint.failAfterError());
 });
 
 /**
- * Copies necessary files for the Babel amd loader to the app.
+ * Copy all non-js directory app source/assets/components.
  */
-gulp.task('loader-polyfill', function() {
-  return gulp.src([
-      './node_modules/alameda/alameda.js',
-      './node_modules/whatwg-fetch/fetch.js',
-      './node_modules/url-search-params/build/url-search-params.max.js',
-      `${APP_ROOT}js/bootstrap.js`,
-    ])
-    .pipe(concat('initapp.js'))
-    .pipe(gulp.dest(`${DIST_APP_ROOT}js`));
+gulp.task('copy-app', function() {
+  return merge(
+    gulp.src([
+      `${APP_ROOT}**`,
+      // Don't copy documentation files.
+      `!${APP_ROOT}**/*.md`,
+      // Don't copy JS, it will be compiled and copied on the compile step.
+      `!${APP_ROOT}js/**`,
+    ]),
+
+    // Module loader.
+    gulp.src('./node_modules/alameda/alameda.js')
+      .pipe(rename('js/alameda.js')),
+
+    // Components.
+    gulp.src('./node_modules/react/dist/react.min.js')
+      .pipe(rename('js/components/react.js')),
+    gulp.src('./node_modules/react-dom/dist/react-dom.min.js')
+      .pipe(rename('js/components/react-dom.js')),
+    gulp.src('./node_modules/fxos-mvc/dist/mvc.js')
+      .pipe(rename('js/components/mvc.js')),
+
+    // Polyfills.
+    gulp.src('./node_modules/whatwg-fetch/fetch.js')
+      .pipe(rename('js/polyfills/fetch.js')),
+    gulp.src('./node_modules/url-search-params/build/url-search-params.js')
+      .pipe(rename('js/polyfills/url-search-params.js'))
+  )
+  .pipe(gulp.dest(DIST_APP_ROOT));
 });
 
 /**
- * Copy all non-js directory app source/assets/components.
+ * Compiles app source i.e. processes source with Bable and Rollup.
  */
-gulp.task('copy-app', function() {
-  return gulp.src([
-      `${APP_ROOT}**`,
-      `!${APP_ROOT}js/**/*.js`, // do not copy js
-    ])
-    .pipe(gulp.dest(DIST_APP_ROOT));
+gulp.task('compile-app', function() {
+  return gulp
+    .src(`${APP_ROOT}js/*.js`)
+    .pipe(
+      rollup({
+        sourceMap: true,
```

Comment:
Now we have es6 sourcemaps! That means that even debug build will have single app.js, but source maps should work well for us - I've played with them for a few days and haven't noticed any troubles. If in the future we encounter any problems, we can disable Rollup for debug build (like we do for unit tests).